Embeded iframe could not refresh when input changes

**Describe the bug**
Embeded iframe could not refresh when input changes。And interesting thing is if you copy the encoded(jwt thing) iframeUrl in any browser,  the input does not worked, but when you refresh the url, and it will work properly again!

**To Reproduce**
1 share any dashborad with any filter by embeded url（the third way）
2 Use the suggested code(I used the python vesion) to encode the shared url
3 embed the encodes url in html files 

**Expected behavior**
shoud refresh as soon as I type some values in the filters

**Information about your Metabase Installation:**
{
  "browser-info": {
    "language": "zh-CN",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.54 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "1.8.0_292-b10",
    "java.vendor": "Red Hat, Inc.",
    "java.vendor.url": "https://www.redhat.com/",
    "java.version": "1.8.0_292",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "25.292-b10",
    "os.name": "Linux",
    "os.version": "3.10.0-1160.el7.x86_64",
    "user.language": "zh",
    "user.timezone": "Asia/Shanghai"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "clickhouse",
      "mysql"
    ],
    "hosting-env": "unknown",
    "application-database": "mysql",
    "application-database-details": {
      "database": {
        "name": "MySQL",
        "version": "8.0.25"
      },
      "jdbc-driver": {
        "name": "MariaDB Connector/J",
        "version": "2.7.5"
      }
    },
    "run-mode": "prod",
    "version": {
      "tag": "v0.43.0",
      "date": "2022-05-02",
      "branch": "release-x.43.x",
      "hash": "ee686fc"
    },
    "settings": {
      "report-timezone": "Asia/Shanghai"
    }
  }
}


**Severity**
its blocking some users

## Steps to reproduce:
1 Create a question by a custom sql query (varialbles are both text type）: 
SELECT * FROM PEOPLE WHERE STATE={{state}} AND CITY={{city}}

2 Create a dashborad, and add the question we made before, which shall give you a result like(Filters are both Text and category type, and dropdown style,  link to the proper varible we've create in the previous step):
![image](https://user-images.githubusercontent.com/25010312/167588038-4079b94b-2ca8-4751-a425-d2ca9f6ef572.png)

3 Pusblish it with both varible editable.

4 Using the iframeUrl-generating code as the site suggests（I used the python version）to generate a iframeUrl.
```python
import jwt
import time

METABASE_SITE_URL = "your ip address"
METABASE_SECRET_KEY = "you token"

payload = {
  "resource": {"dashboard": your dashborad id},
  "params": {"state":"AL"},
  "exp": round(time.time()) + (60 * 10) # 10 minute expiration
}
token = jwt.encode(payload, METABASE_SECRET_KEY, algorithm="HS256")

iframeUrl = METABASE_SITE_URL + "/embed/dashboard/" + token + "#bordered=true&titled=true"
```
5 Copy the  iframeUrl to the browser, you can see the result like below:
![image](https://user-images.githubusercontent.com/25010312/167590384-fe2982ad-1f6c-4ecc-b9a5-35a5a5208053.png)
It didn't work at all 
6 Bizarre thing is refreshing the url, it will work again!
![image](https://user-images.githubusercontent.com/25010312/167591197-f0bfbf82-d17d-45f5-a141-6ac0e2ed536f.png)

## Some web browser logs may help
```
settings.js:110 Error getting setting table.columns TypeError: Cannot read properties of undefined (reading 'cols')
    at Object.getDefault (Table.jsx:182:11)
    at bt (settings.js:101:39)
    at vt (settings.js:42:5)
    at b (visualization.js:77:10)
    at o.value (Visualization.jsx:150:9)
    at o.value (Visualization.jsx:81:10)
    at Lo (react-dom.production.min.js:137:128)
    at Ba (react-dom.production.min.js:176:159)
    at xs (react-dom.production.min.js:263:490)
    at yl (react-dom.production.min.js:246:265)
```

--- a/frontend/src/metabase/dashboard/utils.js
+++ b/frontend/src/metabase/dashboard/utils.js
@@ -85,7 +85,7 @@ export async function fetchDataOrError(dataPromise) {
   }
 }
 
-export function getDatasetQueryParams(datasetQuery) {
+export function getDatasetQueryParams(datasetQuery = {}) {
   const { type, query, native, parameters = [] } = datasetQuery;
   return { type, query, native, parameters };
 }

--- a/frontend/test/metabase/scenarios/sharing/reproductions/22524-public-dashboard-updates-after-changing-parameters.cy.spec.js
+++ b/frontend/test/metabase/scenarios/sharing/reproductions/22524-public-dashboard-updates-after-changing-parameters.cy.spec.js
@@ -1,0 +1,68 @@
+import {
+  restore,
+  popover,
+  visitDashboard,
+  saveDashboard,
+  editDashboard,
+  setFilter,
+} from "__support__/e2e/cypress";
+
+const questionDetails = {
+  name: "22524 question",
+  native: {
+    query: "select * from people where city = {{city}}",
+    "template-tags": {
+      city: {
+        id: "6d077d39-a420-fd14-0b0b-a5eb611ce1e0",
+        name: "city",
+        "display-name": "City",
+        type: "text",
+      },
+    },
+  },
+};
+
+describe("issue 22524", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("update dashboard cards when changing parameters on publicly shared dashboards (metabase#22524)", () => {
+    cy.createNativeQuestionAndDashboard({ questionDetails }).then(
+      ({ body: { dashboard_id } }) => {
+        visitDashboard(dashboard_id);
+      },
+    );
+
+    editDashboard();
+    setFilter("Text or Category", "Dropdown");
+
+    cy.findByText("Select…").click();
+    popover()
+      .contains("City")
+      .click();
+
+    saveDashboard();
+
+    // Share dashboard
+    cy.icon("share").click();
+    cy.findByRole("switch").click();
+
+    cy.findByText("Public link")
+      .parent()
+      .within(() => {
+        cy.get("input").then(input => {
+          cy.visit(input.val());
+        });
+      });
+
+    // Set parameter value
+    cy.findByPlaceholderText("Text")
+      .clear()
+      .type("Rye{enter}");
+
+    // Check results
+    cy.findByText("2-7900 Cuerno Verde Road");
+  });
+});

